### DISCUSSION TITLE
develop-reactjs-asp-net-web-api-apps-in-visual-studio-2015/

### DISCUSSION THREAD

**Discussion by martijnboland** [martijnboland/blog#51] [Blog comments]
# develop-reactjs-asp-net-web-api-apps-in-visual-studio-2015/

https://blogs.taiga.nl/martijn/posts/develop-reactjs-asp-net-web-api-apps-in-visual-studio-2015/

**Comment by disqus-comments-migrator** [comment]
<em>[Ievgen Goichuk](https://disqus.com/by/bobsyourevgeny/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at januari 22 2016, 12:52</em>

---

<p>You are running two dev web-servers in this setup: IISExpress to host WebAPI and web-pack server to host web-pages. You could have client app &amp; assets in WebAPI project and IISExpress also supports hot replacement. What do you achieve with two projects, better isolation, be able to deploy independently and easier swap back-end?</p>

**Reply by martijnboland** [reply]
<em>[Martijn Boland](https://disqus.com/by/martijnboland/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at januari 22 2016, 01:05, in reply to [Ievgen Goichuk](https://disqus.com/by/bobsyourevgeny/)'s [comment](https://github.com/martijnboland/blog/discussions/51#discussioncomment-8952953)</em>

---

<p>Better isolation. The web project is set up just like any other webpack-react project and can still be developed without visual studio. I strongly think that api development and client development should be separate because there is a physical boundary between these two (the network).</p><p>Besides that, I don't think you can do hot module replacement with IISExpress just like with the webpack server. Like editing a js file and immediately see the changes reflected in the browser without losing any state</p>

**Reply by disqus-comments-migrator** [reply]
<em>[Mark Pieszak](https://disqus.com/by/MarkPieszak/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at februari 29 2016, 04:34, in reply to [Ievgen Goichuk](https://disqus.com/by/bobsyourevgeny/)'s [comment](https://github.com/martijnboland/blog/discussions/51#discussioncomment-8952953)</em>

---

<p>How do you set up IIS EXPRESS with hot module replacement?? Didn't know it was possible! Do you have any links or demo github repository of doing this? I'd love to be able to use webpack with hmr and IIS!</p>

**Reply by disqus-comments-migrator** [reply]
<em>[Mark Pieszak](https://disqus.com/by/MarkPieszak/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at februari 29 2016, 04:35, in reply to [Martijn Boland](https://disqus.com/by/martijnboland/)'s [comment](https://github.com/martijnboland/blog/discussions/51#discussioncomment-8952955)</em>

---

<p>Oh its not possible to do HMR with IIS express? =( I got excited for a second. Yet another reason why Node wins in my book =/</p>

**Comment by disqus-comments-migrator** [comment]
<em>[José Señaris](https://disqus.com/by/jossearis/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at januari 25 2016, 09:26</em>

---

<p>Hello... I have just this very same problem. I need to develop a react app, with webpack and all the usual stuff but I'm bound to use a .net backend. This seems promising, just one question.. do i need to set up any extra hooks to run webpack scripts for deploy or i can just use the webpack extension for VS...?</p>

**Reply by martijnboland** [reply]
<em>[Martijn Boland](https://disqus.com/by/martijnboland/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at januari 25 2016, 10:08, in reply to [José Señaris](https://disqus.com/by/jossearis/)'s [comment](https://github.com/martijnboland/blog/discussions/51#discussioncomment-8952959)</em>

---

<p>To make a deployment package, you can use the webpack task runner extension in VS with 'Run - Production' and an optional production webpack config file, see also the example project (webpack.prod.config.js, the output goes to the /dist folder).<br>For build server scenarios, it good to know that you don't need Visual Studio. Just use the webpack or npm command line tools directly.</p>

**Comment by disqus-comments-migrator** [comment]
<em>[mx](https://disqus.com/by/disqus_NZBdChLs76/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at maart 26 2016, 04:17</em>

---

<p>Thank you for this great post! Being a little bit spoilt by the previous Visual Studio Experience (New Project -&gt; F5 and it just runs) i'm a bit overwhelmed by the lots of different little tools and options, Web Development outside the Microsoft Universe offers. So i'm wondering if, when using the Webpack Development Server - is it still possible to do Full Client-to-Server Debugging within Visual Studio? I only saw i can debug the webpacked Javascript Output but this is not what i want as i would like to debug the original Source. Is there an option to omit this packing for Development purposes? Could i still run this Project with IIS Express only? Thank you for your help!</p>

**Reply by martijnboland** [reply]
<em>[Martijn Boland](https://disqus.com/by/martijnboland/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at maart 27 2016, 01:32, in reply to [mx](https://disqus.com/by/disqus_NZBdChLs76/)'s [comment](https://github.com/martijnboland/blog/discussions/51#discussioncomment-8952961)</em>

---

<p>I think full client-to-server debugging in VS is not possible. However, when using the chrome debugger, it is still possible to debug the original sources but these are a little bit hidden in the webpack://. section. Webpack also has a number of 'devtool' options to determine if and how sourcemaps are generated.</p>

**Comment by disqus-comments-migrator** [comment]
<em>[sbanwart](https://disqus.com/by/sbanwart/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at maart 28 2016, 08:40</em>

---

<p>Are there any advantages to using React with Visual Studio this way vs. using ReactJS.NET?</p>

**Reply by martijnboland** [reply]
<em>[Martijn Boland](https://disqus.com/by/martijnboland/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at maart 28 2016, 10:25, in reply to [sbanwart](https://disqus.com/by/sbanwart/)'s [comment](https://github.com/martijnboland/blog/discussions/51#discussioncomment-8952963)</em>

---

<p>Yes, you don't need the extra NuGet packages and it still allows you to develop the client ReactJS apps with only the command-line tooling without Visual Studio. <br>I guess that if you want to do server-side rendering with .NET, ReactJS.NET is the better option though.</p>

**Reply by disqus-comments-migrator** [reply]
<em>[sbanwart](https://disqus.com/by/sbanwart/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at maart 28 2016, 10:50, in reply to [Martijn Boland](https://disqus.com/by/martijnboland/)'s [comment](https://github.com/martijnboland/blog/discussions/51#discussioncomment-8952966)</em>

---

<p>Good. That's what I was thinking too. Thanks again for this great article. It's been very helpful for a JavaScript/React novice like me. :-)</p>

**Reply by disqus-comments-migrator** [reply]
<em>[Paul Knopf](https://disqus.com/by/pauldotknop/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at mei 30 2016, 08:28, in reply to [sbanwart](https://disqus.com/by/sbanwart/)'s [comment](https://github.com/martijnboland/blog/discussions/51#discussioncomment-8952963)</em>

---

<p>ReactJS.NET does too much magic for what little it should be doing.</p><p>Check the react-aspnet-boilerplate. <a href="https://github.com/pauldotknopf/react-aspnet-boilerplate" rel="nofollow noopener" title="https://github.com/pauldotknopf/react-aspnet-boilerplate">https://github.com/pauldotknopf/react-aspnet-boilerplate</a></p>

**Comment by disqus-comments-migrator** [comment]
<em>[Thomas](https://disqus.com/by/disqus_RU0dNycJDA/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at mei 11 2016, 04:52</em>

---

<p>Has anybody got this working with ASP.NET 5/ASP.NET Core? I want to be able to develop in VSCode on my Mac/OSX (and other platforms)? Do all the tools work?</p>

**Reply by martijnboland** [reply]
<em>[Martijn Boland](https://disqus.com/by/martijnboland/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at mei 11 2016, 12:11, in reply to [Thomas](https://disqus.com/by/disqus_RU0dNycJDA/)'s [comment](https://github.com/martijnboland/blog/discussions/51#discussioncomment-8952970)</em>

---

<p>This blog is explicitly about Visual Studio. When using VSCode, I'd just use the command-line tooling.</p>

**Reply by disqus-comments-migrator** [reply]
<em>[Paul Knopf](https://disqus.com/by/pauldotknop/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at mei 30 2016, 08:27, in reply to [Thomas](https://disqus.com/by/disqus_RU0dNycJDA/)'s [comment](https://github.com/martijnboland/blog/discussions/51#discussioncomment-8952970)</em>

---

<p>Better use the react-aspnet-boilerplate template for ASP.NET Core, React and server-side rendering.</p><p><a href="https://github.com/pauldotknopf/react-aspnet-boilerplate" rel="nofollow noopener" title="https://github.com/pauldotknopf/react-aspnet-boilerplate">https://github.com/pauldotknopf/react-aspnet-boilerplate</a></p>

**Comment by disqus-comments-migrator** [comment]
<em>[Mihir Karandikar](https://disqus.com/by/mihirkarandikar/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at juli 13 2016, 10:46</em>

---

<p>Hi there! Can you provide some docs for running my react+webpack project in production mode? I ran the webpack.prod.config.js, and it dumped a bundle inside "/dist". But what do I do after that? Also, do I have to change localhost:3000 to something else in project properties? Thanks for the awesome tutorial though! It has saved me a lot of time :)</p>

**Reply by martijnboland** [reply]
<em>[Martijn Boland](https://disqus.com/by/martijnboland/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at juli 13 2016, 11:03, in reply to [Mihir Karandikar](https://disqus.com/by/mihirkarandikar/)'s [comment](https://github.com/martijnboland/blog/discussions/51#discussioncomment-8952973)</em>

---

<p>The bundle is just a static website that you can upload to your production environment or even run directly from the filesystem.</p>

**Reply by disqus-comments-migrator** [reply]
<em>[Mihir Karandikar](https://disqus.com/by/mihirkarandikar/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at juli 13 2016, 11:38, in reply to [Martijn Boland](https://disqus.com/by/martijnboland/)'s [comment](https://github.com/martijnboland/blog/discussions/51#discussioncomment-8952975)</em>

---

<p>Thank you for the prompt response. I have some queries. What is the best way to server that static website? I'm sorry, I'm completely new to .NET environment. Do I create a new project that serves these static files inside VSReact solution? In node, I'd create an express server to serve these files. Also, why am I not able to see "build", "dist", "bin", "node_modules" folders inside VSReact.Web project tree? Is there something like .gitignore thing working behind the scenes?</p>

**Comment by disqus-comments-migrator** [comment]
<em>[Aston Dewy](https://disqus.com/by/nisismlevy/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at november 02 2016, 12:44</em>

---

<p>I can't find the link to go to the step by step instructions of this tutorial</p>

**Reply by martijnboland** [reply]
<em>[Martijn Boland](https://disqus.com/by/martijnboland/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at november 02 2016, 08:56, in reply to [Aston Dewy](https://disqus.com/by/nisismlevy/)'s [comment](https://github.com/martijnboland/blog/discussions/51#discussioncomment-8952978)</em>

---

<p>I'm not sure I understand the question. What link do you mean?</p>

**Reply by disqus-comments-migrator** [reply]
<em>[Nissim Levy](https://disqus.com/by/nissim_levy/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at november 02 2016, 10:08, in reply to [Martijn Boland](https://disqus.com/by/martijnboland/)'s [comment](https://github.com/martijnboland/blog/discussions/51#discussioncomment-8952979)</em>

---

<p>Where is the tutorial? You know...lesson 1, lesson 2 etc..</p>

**Reply by martijnboland** [reply]
<em>[Martijn Boland](https://disqus.com/by/martijnboland/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at november 03 2016, 09:56, in reply to [Nissim Levy](https://disqus.com/by/nissim_levy/)'s [comment](https://github.com/martijnboland/blog/discussions/51#discussioncomment-8952982)</em>

---

<p>There is no tutorial. This is just a blog post.</p>
